File: prog3.cpp
/*
 * Name: Shawn Picardy
 * Course: CPSC 3220
 * Program 3
 * March 24
 */

// arrival time and service time input file. example: 3 4 from instruction page.
#include <iostream>
#include <stdlib.h>
#include <vector>
#include <string>
using namespace std;

vector<int> arrivalTime;
vector<int> serviceTime;

void printScheduling(string a)
{
    // additional string versions per the examples
    string rrSuffix = "(time slice is 1)";
    string sjfSuffix = "(preemptive)";

    //Convert incoming string to uppercase
    transform(a.begin(), a.end(), a.begin(), ::toupper);

    //Print appropriate heading per string parameter
    if (a == "FIFO")
    {
        cout << "\n" + a + " scheduling results" << endl;
    }
    else if(a == "SJF"){
        cout << "\n" + a + sjfSuffix + " scheduling results" << endl;
    }
    else if (a == "RR"){
        cout << "\n" + a + " scheduling results " + rrSuffix << endl;
    }
    else
    {
        cout << "AN ERRROR OCCURED" << endl;
    }
    
}

int main(int argc, char const *argv[])
{

    printScheduling(argv[1]);
    return 0;
}

File: prog3.java
/**
 * Name: Shawn Picardy
 * Course: cpsc 3220
 * Smotherman
 * March 17th. 
 */

import java.util.Scanner;
import java.util.Vector;

class Schedule {

    private static Vector<Integer> arrivalTime = new Vector<Integer>();
    private static Vector<Integer> serviceTime = new Vector<Integer>();

    private static int remainingTime; 
    private static int completionTime;
    private static int responseTime;
    private static int waitTime;  

    // int
    // task_id, /* alphabetic tid can be obtained as 'A'+(task_counter++) */
    // arrival_time,
    // service_time,
    // remaining_time,
    // completion_time,
    // response_time,
    // wait_time;

    public static void main(String[] args) {
        acquireTimes();
        printTraceTimes();
    }

    private static void acquireTimes() {
        // Create scanner for I/O redirection
        Scanner input = new Scanner(System.in);
        // Use a counter and modulus to retreive each value from stdin
        // then place every other value into respective vectors.
        int counter = 0;
        while (input.hasNext()) {
            String a = input.next();
            if ((counter % 2) == 0) {
                arrivalTime.add(Integer.parseInt(a));
            } else {
                serviceTime.add(Integer.parseInt(a));
            }
            counter++;
        }
        input.close(); // close scanner to prevent leak

        // convert both vectors to strings and print for debugging.
        arrivalTime.toString();
        serviceTime.toString();

        /** ARRIVAL TIMES */
        System.out.println("Arrival times are:");
        for (Integer element : arrivalTime) {
            System.out.print(element + " ");
        }
        System.out.print("\n\r");

        /** SERVICE TIMES */
        System.out.println("Service times are:");
        for (Integer element : serviceTime) {
            System.out.print(element + " ");
        }
        System.out.print("\n\r");

    }

    private static void printTraceTimes() {
        System.out.print("time   cpu   ready queue (tid/rst)\n----   ---   ---------------------\n\r");
    }

    private static void printTaskSummary() {
        System.out.print("Task summary table");
    }

    private static void printTimeSummary() {
        System.out.print("Time tracing");
    }
}
